vendor/bat-native-ledger/src/bat/ledger/internal/logging/logging.cc

```diff
@@ -5,28 +5,29 @@
 
 #include "bat/ledger/internal/logging/logging.h"
 
+#include "base/lazy_instance.h"
+#include "base/threading/thread_local.h"
 #include "bat/ledger/ledger_client.h"
 
 namespace ledger {
 
-LedgerClient* g_ledger_client = nullptr;  // NOT OWNED
```

Comment:
There is a problem with this static: it doesn't synchronized (i.e. guarded by a mutex or else), so it's unsafe to use it from multiple threads now.
In addition, with enabled in-processing, the browser process also starts to use it (also from another thread).

I made it thread-local as a preliminary solution.

Reply:
why would this be accessed from more than one thread? If that's happening now then I think there's an existing bug

Reply:
The ledger process has (unfortunately) used this global variable since logging was introduced in 2020. Clearly, the change then assumed that `bat-native-ledger` would be run in its own process.

Reply:
Does ledger inside uses other threads or not? Can it this var be used from another thread now (in case of dedicated process)?

I've also found that the main process also is trying to write some logs, but fails because `!g_ledger_client`.
After making impl in-process the logging starts working, but it can lead to data race on `g_ledger_client`.

Reply:
It looks like `g_ledger_client` is used via `BLOG(...)` in `ledger_database_impl.cc`, which is run from a thread in the browser process. I think `bat-native-ads` is set up the same way.

Reply:
@atuchin-m can we split out just this part from the rest since it's an issue independent of in vs out of process? This PR will likely have to change anyway because of the PR to combine ledger/ads. However, I don't see why we still can't add a feature flag for in process so we can start working in that direction